Traktor S3: Disable scratch when switching decks to prevent locked scratch issue

evasadakkanalesso opened the pull request:
Hi there,

I have been using the Traktor S3 controller mappings and have come across this sneaky bug. Steps to reproduce below:

1. Start any song you like and perform a spin in whichever direction you feel like
2. quickly swap decks and you will find the previous deck is kept in scratch mode

This happens due to ```TraktorS3.Deck.prototype.checkJogInertia``` function using ```this.activeChannel``` which is now different than the previous active channel. Traktor DJ Pro works the same way as in if you perform a spin in any direction and swap decks it interrupts the current spinning deck. 

Thanks Mixxx Dev for building such an amazing software!!!!

Thanks,
Eva

Commit:
Disable scratch when switching decks to prevent infinite scratch issue

Be-ing commented:
Thanks. I have retargeted this PR to the 2.3 branch. Please [rebase it](https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch).

evasadakkanalesso commented:
Hi @Be-ing ,

It's done. 

👍

Be-ing commented:
Thank you. @ywwg wrote the mapping, so I'll leave it to him to review and test these changes.

evasadakkanalesso committed:
Update Traktor-Kontrol-S3-hid-scripts.js

Hello everyone,

Just realised the code above did not work. It worked for me because I use a modified version of this script.

Thanks

ywwg commented on res/controllers/Traktor-Kontrol-S3-hid-scripts.js:
```diff
@@ -1794,6 +1796,12 @@ TraktorS3.Controller.prototype.deckSwitchHandler = function(field) {
     var deck = channel.parentDeck;
     deck.activateChannel(channel);
+
+    var isScratching = engine.getValue(deck.previousActiveChannel, "scratch2_enable");
```
instead of this new "previouschannel" value, why not disable scratching at the top of the deckswitchhandler, before the active channel gets updated?  so, disable scratching and then activateChannel.

evasadakkanalesso replied:
hi @ywwg - that's doable and also an easier way of doing it. Will make the change 👍 
thank you

evasadakkanalesso committed:
Code change as per requested change

Hi @ywwg,

I have made the code change and tested on my machine and it seems to be working fine.

Thank you 👍

uklotzde commented:
@evasadakkanalesso As a *first-time contributor* we have to ask you to sign our contributor agreement: https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

@ywwg Please check before merging PRs.

ywwg commented:
ah, yes, sorry.

ywwg commented:
@evasadakkanalesso Hi, please let us know when you've signed the agreement -- otherwise we'll have to roll back this change because we need every contribution logged.

evasadakkanalesso commented:
Hi Owen,

I have done it a few days. I thought you'd get a message or something.

Thanks
On Thu, Jul 29, 2021 at 2:07 PM Owen Williams ***@***.***>
wrote:

> @evasadakkanalesso <https://github.com/evasadakkanalesso> Hi, please let
> us know when you've signed the agreement -- otherwise we'll have to roll
> back this change because we need every contribution logged.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/4073#issuecomment-889127818>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUW7MCFZJD3PIKNCZYPW32TT2FHA3ANCNFSM477QXO6Q>
> .
>

ywwg commented:
Thanks!

evasadakkanalesso commented:
no problem thank you! ♥️

On Fri, Jul 30, 2021 at 3:13 PM Owen Williams ***@***.***>
wrote:

> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/4073#issuecomment-889921020>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUW7MCA5HVUO6UOSPB7F6QDT2KXO5ANCNFSM477QXO6Q>
> .
>
